src/Desktop.Analyzers/Core/DoNotUseInsecureDTDProcessing.cs

```diff
@@ -367,475 +326,520 @@ private void AnalyzeNodeForDtdProcessingOverloads(SyntaxNodeAnalysisContext cont
+                {
+                    AnalyzeObjectCreationForXmlReaderSettings(context, variable, objCreation);
+                }
+                else
+                {
+                    AnalyzeMethodOverloads(context, objCreation.Constructor, objCreation.ArgumentsInParameterOrder, objCreation.Syntax.GetLocation());
                 }
             }
 
-            private void AnalyzeNodeForXmlDocument(SyntaxNodeAnalysisContext context)
+            private void AnalyzeObjectCreationForXmlDocument(OperationAnalysisContext context, ISymbol variable, IObjectCreationExpression objCreation)
             {
-                SyntaxNode node = context.Node;
-                SemanticModel model = context.SemanticModel;
-
-                node = _syntaxNodeHelper.GetVariableDeclaratorOfAFieldDeclarationNode(node) ?? node;
-
-                SyntaxNode lhs = _syntaxNodeHelper.GetAssignmentLeftNode(node);
-                SyntaxNode rhs = _syntaxNodeHelper.GetAssignmentRightNode(node);
+                XmlDocumentEnvironment env = null;
 
-                if (lhs == null || rhs == null)
+                if (variable == null || !_xmlDocumentEnvironments.ContainsKey(variable))
                 {
-                    return;
+                    env = new XmlDocumentEnvironment();
+                    env.IsSecureResolver = false;
+                    env.IsXmlResolverSet = false;
+                    env.XmlDocumentDefinition = objCreation.Syntax;
+                }
+                else
+                {
+                    env = _xmlDocumentEnvironments[variable];
+                    env.XmlDocumentDefinition = objCreation.Syntax;
                 }
 
-                ISymbol lhsSymbol = SyntaxNodeHelper.GetSymbol(lhs, model);
-                if (lhsSymbol == null)
+                SyntaxNode node = objCreation.Syntax;
+                bool isXmlDocumentSecureResolver = false;
+
+                if (objCreation.Constructor.ContainingType != _xmlTypes.XmlDocument)
                 {
-                    return;
+                    isXmlDocumentSecureResolver = true;
                 }
 
-                CompilationSecurityTypes xmlTypes = _xmlTypes;
-                IMethodSymbol rhsMethodSymbol = _syntaxNodeHelper.GetCalleeMethodSymbol(rhs, model);
-                if (SecurityDiagnosticHelpers.IsXmlDocumentCtorDerived(rhsMethodSymbol, xmlTypes))
+                foreach (ISymbolInitializer init in objCreation.MemberInitializers)
                 {
-                    XmlDocumentEnvironment env = new XmlDocumentEnvironment();
+                    var prop = init as IPropertyInitializer;
 
-                    if (rhsMethodSymbol.ContainingType != xmlTypes.XmlDocument)
+                    if (prop != null)
                     {
-                        env.IsSecureResolver = true;
-                    }
+                        if (prop.InitializedProperty.MatchPropertyDerivedByName(_xmlTypes.XmlDocument, "XmlResolver"))
```

Comment:
Method `MatchPropertyDerivedByName` does not need the check `member.Kind == SymbolKind.Property `